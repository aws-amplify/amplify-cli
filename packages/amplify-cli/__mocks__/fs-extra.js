const fsExtra = jest.genMockFromModule('fs-extra');

function copySync() {
  return {};
}

fsExtra.copySync = copySync;

module.exports = fsExtra;
