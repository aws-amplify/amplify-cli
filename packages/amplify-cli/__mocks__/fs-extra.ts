const fs: any = jest.genMockFromModule('fs-extra');

fs.readdirSync = jest.fn().mockReturnValue(['file1', 'file2']);
fs.readFileSync = jest.fn().mockReturnValue({});
fs.statSync = jest.fn().mockReturnValue({ isDirectory: () => {} });
fs.unlinkSync = jest.fn().mockReturnValue(null);
fs.copySync = jest.fn().mockReturnValue({});

module.exports = fs;
