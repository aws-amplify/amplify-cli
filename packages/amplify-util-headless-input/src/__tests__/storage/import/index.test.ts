import * as fs from 'fs-extra';
import path from 'path';
import { validateImportStorageRequest } from '../../..';

const importStorageAssetsRoot = path.resolve(path.join(__dirname, '../../assets/storage/import'));

type ExpectedFunction = (missingProperty: string) => string;

describe('validates import storage headless request', () => {
  const topLevelMissingProperty: ExpectedFunction = (missingProperty: string) =>
    `Data did not validate against the supplied schema. Underlying errors were [{\"keyword\":\"required\",\"dataPath\":\"\",\"schemaPath\":\"#/required\",\"params\":{\"missingProperty\":\"${missingProperty}\"},\"message\":\"should have required property '${missingProperty}'\"}]`;
  const serviceConfigurationMissingProperty: ExpectedFunction = (missingProperty: string) =>
    `Data did not validate against the supplied schema. Underlying errors were [{\"keyword\":\"required\",\"dataPath\":\".serviceConfiguration\",\"schemaPath\":\"#/definitions/ImportS3ServiceConfiguration/required\",\"params\":{\"missingProperty\":\"${missingProperty}\"},\"message\":\"should have required property '${missingProperty}'\"}]`;

  it('valid payload passes validation', async () => {
    const rawRequest = fs.readFileSync(path.join(importStorageAssetsRoot, 'valid.importStorageRequest.json'), 'utf8');
    const result = await validateImportStorageRequest(rawRequest);
    expect(result).toMatchSnapshot();
  });

  it('non-json content rejects', async () => {
    const resultPromise = validateImportStorageRequest('garbage');
    expect(resultPromise).rejects.toBeTruthy();
  });

  test.each([
    [
      'version',
      'invalidRequest.version.missing.json',
      (missingProperty: string) => `data does not have a top level \"${missingProperty}\" field`,
    ],
    [
      'version',
      'invalidRequest.version.string.json',
      (missingProperty: string) => `data does not have a top level \"${missingProperty}\" field`,
    ],
    ['version', 'invalidRequest.version.invalid.json', (missingProperty: string) => `No schema found for ${missingProperty} 9999`],
    ['serviceConfiguration', 'invalidRequest.missing.serviceConfiguration.json', topLevelMissingProperty],
    ['serviceName', 'invalidRequest.missing.serviceName.json', serviceConfigurationMissingProperty],
    ['bucketName', 'invalidRequest.missing.bucketName.json', serviceConfigurationMissingProperty],
  ])(
    'with a rejected promise when the payload is missing %s',
    async (missingProperty: string, fileName: string, expectedFn: ExpectedFunction) => {
      const rawRequest = fs.readFileSync(path.join(importStorageAssetsRoot, fileName), 'utf8');

      await expect(validateImportStorageRequest(rawRequest)).rejects.toThrow(expectedFn(missingProperty));
    },
  );
});
