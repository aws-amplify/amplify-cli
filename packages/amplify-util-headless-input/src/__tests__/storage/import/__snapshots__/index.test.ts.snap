// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validates import storage headless request valid payload passes validation 1`] = `
Object {
  "serviceConfiguration": Object {
    "bucketName": "import-this-bucket",
    "serviceName": "S3",
  },
  "version": 1,
}
`;
